crates/net/network/src/peers/manager.rs

```diff
@@ -251,7 +251,16 @@ impl PeersManager {
     /// Temporarily puts the peer in timeout
     fn backoff_peer(&mut self, peer_id: PeerId, kind: BackoffKind) {
```

Comment:
this is only called once in `on_connection_failure` where we already have looked up the peer.

so I'd suggest rewriting this to `backoff_peer_until(PeerId, Instant)` and we compute the Instant before we call this via `backoff_until`

Reply:
Okay, will work on it. Thanks!

Reply:
Could you please elaborate on the `backoff_peer_until(PeerId, Instant)` function?  Should this be a new function where we move the new logic from `backoff_peer`?

I'm also not sure where we will look up the `backoff_counter`. Because in the `on_connection_failure` function, we look it up after we back off the peer. https://github.com/paradigmxyz/reth/blob/main/crates/net/network/src/peers/manager.rs#L390

Should I switch the looking up of the peer with the backoff mechanism?

Thanks a lot!

Reply:
> Because in the on_connection_failure function, we look it up after we back off the peer.

right, we can move everything inside the `if let` 

https://github.com/paradigmxyz/reth/blob/49132a8d1af13646c94ecb45c61644468800443c/crates/net/network/src/peers/manager.rs#L390

compute the backoff inside, update the counter, then call `backoff_peer_until `

perhaps there are some borrow checker issues if you try to call `self.backoff_peer_until` inside the if let but you can do something like

```
let mut backoff_until = None
if let Some(mut peer) = self.peers.get_mut(peer_id) {
 // --snip--
 if should_backoff() {
   backoff_until = ...;
  }
}

if let Some(backoff_until) = backoff_until {
  /....
}

```

and we need to change what's currently called `backoff_peer` to `backoff_peer_until` so it already takes the Instant

does this unblock you?

Reply:
Yes makes sense now. Thanks a lot!